test/e2e/tests/onboarding/onboarding.spec.js

```diff
@@ -301,4 +307,140 @@ describe('MetaMask onboarding @no-mmi', function () {
       },
     );
   });
+
+  it("doesn't make any network requests to infura before onboarding is completed", async function () {
+    async function mockInfura(mockServer) {
+      const infuraUrl =
+        'https://mainnet.infura.io/v3/00000000000000000000000000000000';
+      const sampleAddress = '1111111111111111111111111111111111111111';
+
+      return [
+        await mockServer
+          .forPost(infuraUrl)
+          .withJsonBodyIncluding({ method: 'eth_blockNumber' })
+          .thenCallback(() => {
+            return {
+              statusCode: 200,
+              json: {
+                jsonrpc: '2.0',
+                id: '1111111111111111',
+                result: '0x1',
+              },
+            };
+          }),
+        await mockServer
+          .forPost(infuraUrl)
+          .withJsonBodyIncluding({ method: 'eth_getBalance' })
+          .thenCallback(() => {
+            return {
+              statusCode: 200,
+              json: {
+                jsonrpc: '2.0',
+                id: '1111111111111111',
+                result: '0x1',
+              },
+            };
+          }),
+        await mockServer
+          .forPost(infuraUrl)
+          .withJsonBodyIncluding({ method: 'eth_getBlockByNumber' })
+          .thenCallback(() => {
+            return {
+              statusCode: 200,
+              json: {
+                jsonrpc: '2.0',
+                id: '1111111111111111',
+                result: {},
+              },
+            };
+          }),
+        await mockServer
+          .forPost(infuraUrl)
+          .withJsonBodyIncluding({ method: 'eth_call' })
+          .thenCallback(() => {
+            return {
+              statusCode: 200,
+              json: {
+                jsonrpc: '2.0',
+                id: '1111111111111111',
+                result: `0x000000000000000000000000${sampleAddress}`,
```

Comment:
this was specific for ens. Removing it and just passing zeros would have been fine. No need to change it, just mentioning it for clarity in the case someone wonders why are we passing here a sample address

Reply:
Thanks for clarifying @seaona. This will be revisited and cleaned up on [this PR](https://github.com/MetaMask/metamask-extension/pull/23005).